src/components/PostDetails.tsx

```diff
@@ -1,117 +1,149 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+interface T {
```

Comment:
it's not good idea to use T as a name for interface cause T is usually used for generic type and it seems confused here